### PR TITLE
Adopt the logrotate cookbook into Sous Chefs

### PR DESCRIPTION
**Pull request by TheREK3R**
# Description

Import this cookbook officially into Sous Chefs

## Issues Resolved

Remove EOL platforms and fix style issues

## Check List

- [X] All tests pass. See TESTING.md for details.
- [X] New functionality includes testing.
- [X] New functionality has been documented in the README if applicable.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    strategy:
+      matrix:
+        os:
+          - 'amazonlinux-2'
+          - 'debian-9'
+          - 'debian-10'
+          - 'centos-7'
+          - 'centos-8'
+          - 'fedora-latest'
+          - 'ubuntu-1604'
```
Comment (ramereth) [inline .github/workflows/ci.yml:49]: Please go ahead and remove Ubuntu 16.04 support.

File: .travis.yml [hunk 2]
```diff
@@ -18,15 +15,16 @@ services: docker
...
+    - INSTANCE=resources-ubuntu-1604
+    - INSTANCE=resources-ubuntu-1804
+    - INSTANCE=resources-centos-6
+    - INSTANCE=resources-centos-7
+    - INSTANCE=resources-debian-9
+    - INSTANCE=resources-debian-10
+    - INSTANCE=resources-fedora-latest
+    - INSTANCE=resources-opensuse-leap
+    - INSTANCE=resources-amazonlinux
+    - CHEF_LICENSE=accept
```
Comment (ramereth) [inline .travis.yml:27]: This whole file needs to be removed. We don't use TravisCI.

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,7 +4,15 @@ This file is used to list changes made in each version of the logrotate cookbook
 
 ## Unreleased
 
-- resolved cookstyle error: attributes/default.rb:32:43 refactor: `ChefRedundantCode/MultiplePlatformChecks`
+### Changed
```
Comment (ramereth) [inline CHANGELOG.md:7]: We're not doing the subsections in the changelog anymore, can you please remove this?

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,7 +4,15 @@ This file is used to list changes made in each version of the logrotate cookbook
...
+
+- Sous Chefs Adoption
+- Update Changelog to Sous Chefs
+- Update Readme to Sous Chefs
+- Update metadata.rb to Sous Chefs
```
Comment (ramereth) [inline CHANGELOG.md:12]: These three can go as it's just covered with the `Sous Chefs Adoption` line.

**Review by ramereth — CHANGES_REQUESTED** [review]
In `kitchen.yml` please re-enable `deprecations_as_errors` and also enable the idempotency checks

File: kitchen.yml [hunk 5]
```diff
@@ -4,27 +4,23 @@ driver:
 provisioner:
   name: chef_zero
   chef_license: accept
-  # We have tests that ensure our deprecated features still
-  # work so we can't turn this on globally.
-  # deprecations_as_errors: true
+  deprecations_as_errors: true
+  enforce_idempotency: true
+  multiple_converge: 2
+
```
Comment (ramereth) [inline kitchen.yml:10]: Please remove the extra space here.

Comment (TheREK3R) [reply kitchen.yml:10]: oof, looked for a space for a bit, and didn't realize you meant newline lol

**Comment by ramereth** [comment]
@TheREK3R can you please add the "Maintainers" section like we have here? https://github.com/sous-chefs/freebsd#maintainers

File: kitchen.yml [hunk 6]
```diff
@@ -4,27 +4,22 @@ driver:
...
-  - name: centos-6
   - name: centos-7
   - name: centos-8
   - name: debian-9
   - name: debian-10
   - name: fedora-latest
   - name: freebsd-12
   - name: opensuse-leap-15
-  - name: ubuntu-16.04
   - name: ubuntu-18.04
```
Comment (ramereth) [inline kitchen.yml:23]: Please add Ubuntu 20.04 here, in dokken and in the GH actions tests

File: kitchen.yml [hunk 7]
```diff
@@ -4,27 +4,22 @@ driver:
...
-    driver_config:
-      box: mvbcoding/awslinux
   - name: amazonlinux-2
-  - name: centos-6
   - name: centos-7
   - name: centos-8
   - name: debian-9
   - name: debian-10
   - name: fedora-latest
   - name: freebsd-12
```
Comment (ramereth) [inline kitchen.yml:21]: It doesn't look like freebsd works nor is it included in metadata.rb. I don't even see it mentioned in the history. Let's remove this platform until someone actually needs it.

**Review by ramereth — CHANGES_REQUESTED** [review]

**Review by ramereth — APPROVED** [review]

**Comment by kitchen-porter** [comment]
Released as: [2.3.0](https://github.com/sous-chefs/logrotate/releases/tag/2.3.0)

**Pull request merged**